packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -166,11 +166,15 @@ export const vModelRadio: ModelDirective<HTMLInputElement> = {
 }
 
 export const vModelSelect: ModelDirective<HTMLSelectElement> = {
-  created(el, binding, vnode) {
+  created(el, { modifiers: { number } }, vnode) {
+    const castToNumber = number
     addEventListener(el, 'change', () => {
       const selectedVal = Array.prototype.filter
         .call(el.options, (o: HTMLOptionElement) => o.selected)
-        .map(getValue)
+        .map((o: HTMLOptionElement) => {
+          const domValue = getValue(o)
+          return castToNumber ? toNumber(domValue) : domValue
+        })
```

Comment:
maybe no need to `castToNumber?` every time in the `.map`.
```ts
const selectedVal = castToNumber ? temp.map(o=>toNumber(getValue(o))) : temp.map(getValue)
```

Reply:
🐂B.